airbyte-config/persistence/src/test/java/io/airbyte/config/persistence/ConfigRepositoryTest.java

```diff
@@ -138,6 +141,101 @@ void testUpdateConnectionState() throws Exception {
+    assertReturnsSourceDefinition(new StandardSourceDefinition().withSourceDefinitionId(SOURCE_DEFINITION_ID));
+  }
+
+  @Test
+  void testSourceDefinitionWithTrueTombstone() throws JsonValidationException, ConfigNotFoundException, IOException {
+    assertReturnsSourceDefinition(new StandardSourceDefinition().withSourceDefinitionId(SOURCE_DEFINITION_ID).withTombstone(true));
+  }
+
+  @Test
+  void testSourceDefinitionWithFalseTombstone() throws JsonValidationException, ConfigNotFoundException, IOException {
+    assertReturnsSourceDefinition(new StandardSourceDefinition().withSourceDefinitionId(SOURCE_DEFINITION_ID).withTombstone(false));
+  }
+
+  void assertReturnsSourceDefinition(final StandardSourceDefinition sourceDefinition)
+      throws ConfigNotFoundException, IOException, JsonValidationException {
+    when(configPersistence.getConfig(ConfigSchema.STANDARD_SOURCE_DEFINITION, SOURCE_DEFINITION_ID.toString(), StandardSourceDefinition.class))
+        .thenReturn(sourceDefinition);
+
+    assertEquals(sourceDefinition, configRepository.getStandardSourceDefinition(SOURCE_DEFINITION_ID));
+  }
+
+  @Test
+  void testSourceDefinitionFromSource() throws JsonValidationException, ConfigNotFoundException, IOException {
+    final UUID sourceId = UUID.randomUUID();
+
+    final SourceConnection source = new SourceConnection()
+        .withSourceId(sourceId)
+        .withSourceDefinitionId(SOURCE_DEFINITION_ID);
+
+    doReturn(source)
+        .when(configRepository)
+        .getSourceConnection(sourceId);
+
+    configRepository.getSourceDefinitionFromSource(sourceId);
+    verify(configRepository).getStandardSourceDefinition(SOURCE_DEFINITION_ID);
+  }
+
+  @Test
+  void testSourceDefinitionsFromConnection() throws JsonValidationException, ConfigNotFoundException, IOException {
+    final UUID sourceId = UUID.randomUUID();
+    final UUID connectionId = UUID.randomUUID();
+
+    final StandardSourceDefinition sourceDefinition = new StandardSourceDefinition()
+        .withSourceDefinitionId(SOURCE_DEFINITION_ID);
+
+    final SourceConnection source = new SourceConnection()
+        .withSourceId(sourceId)
+        .withSourceDefinitionId(SOURCE_DEFINITION_ID);
+
+    final StandardSync connection = new StandardSync()
+        .withSourceId(sourceId)
+        .withConnectionId(connectionId);
+
+    doReturn(sourceDefinition)
+        .when(configRepository)
+        .getStandardSourceDefinition(SOURCE_DEFINITION_ID);
+    doReturn(source)
+        .when(configRepository)
+        .getSourceConnection(sourceId);
+    doReturn(connection)
+        .when(configRepository)
+        .getStandardSync(connectionId);
+
+    configRepository.getSourceDefinitionFromSource(sourceId);
+
+    verify(configRepository).getStandardSourceDefinition(SOURCE_DEFINITION_ID);
+  }
+
+  @ParameterizedTest
+  @ValueSource(ints = {0, 1, 2, 10})
+  void testListStandardSourceDefinitions_handlesTombstoneSourceDefinitions(final int numSourceDefinitions)
```

Comment:
Oh, this is a habit/convention that my former team used for unit tests. The idea was to name tests like `<thingBeingTested>_<expectedResult>`. It's useful when there are multiple tests for the same method so that you can quickly parse what behaviors are being tested for a given thing. In this case, I sort of just did it out of habit and it isn't particularly useful since there aren't other unit tests for this method that need differentiating